controllers/reservation.controllers.js

```diff
@@ -83,47 +81,473 @@ const listReservationByDate = async ({ client, ack, say, body }) => {
+
+  return rooms.map((room) => ({
+    text: {
+      type: 'plain_text',
+      text: room.name,
       emoji: true,
     },
-  ];
+    value: room._id,
+  }));
+};
+
+const checkRoomExistence = async (slackId, say, roomName) => {
+  const user = await User.findOne({ slackId });
+  if (!user) {
+    say(':scream: - Error: User not found.');
+    return;
+  }
+  const office = await Office.findById(user.office).populate(
+    'rooms',
+    'name _id',
+  );
+
+  if (!office) {
+    say(':scream: - Error: Office not found.');
+    return;
+  }
+
+  const { rooms } = office;
+
+  return rooms.find((room) => room.name === roomName);
+};
+
+const submitReserve = async ({
+  ack, body, view, client,
+}) => {
+  let errors = {};
+
+  const { date_input, site_input, frecuency_input } = view.state.values;
+
+  const date = date_input['datepicker-action'].selected_date;
+  const room = site_input['static_select-action'].selected_option;
+  const frecuency = frecuency_input['radio_button-action'].selected_option;
+  const slackId = body.user.id;
+
+  const user = await User.findOne({ slackId });
+  const currentDate = moment();
+  const invalidDate = !date || moment(date).diff(currentDate, 'days') < 0;
+  const currentReservations = await Reservation.find({
+    date,
+    office: user.office,
+    room: room.value,
+  })
+    .populate('room', 'enabled maxCapacity')
+    .populate('office', 'enabled maxVisitsAWeek');
+  const isRoomFull = currentReservations.length
+    && currentReservations.length >= currentReservations[0].room.maxCapacity;
+
+  const startOfWeek = moment(date).startOf('isoWeek');
+  const endOfWeek = moment(date).endOf('isoWeek');
+
+  const query = { date: { $gt: startOfWeek, $lt: endOfWeek }, user: user._id };
+  const currentOffice = await Office.findById(user.office);
+  const alreadyHaveReserve = await Reservation.find({
+    date,
+    office: user.office,
+    room: room.value,
+    user: user._id,
+  });
+
+  const weekResevationByUser = await Reservation.find(query);
+
+  if (invalidDate) {
+    errors = {
+      response_action: 'errors',
+      errors: {
+        date_input: 'La fecha no es válida',
+      },
+    };
+  }
+
+  if (!room) {
+    errors = {
+      response_action: 'errors',
+      errors: {
+        site_input: 'Seleccione un lugar',
+      },
+    };
+  }
+
+  if (!frecuency) {
+    errors = {
+      response_action: 'errors',
+      errors: {
+        date_input: 'El frecuencia no es válida',
+      },
+    };
+  }
+
+  if (isRoomFull) {
+    errors = {
+      response_action: 'errors',
+      errors: {
+        site_input: 'No hay lugares disponible en esta sala',
+      },
+    };
+  }
+
+  if (alreadyHaveReserve.length) {
+    errors = {
+      response_action: 'errors',
+      errors: {
+        date_input: 'Ya tienes una reserva para este día',
+      },
+    };
+  }
+
+  if (weekResevationByUser.length > currentOffice.maxVisitsAWeek) {
+    errors = {
+      response_action: 'errors',
```

Comment:
if response_action is always 'errors', why not define it in the variable init and then just update the errors nested object? something like:
`const errors = { response_action: 'errors', errors: {} }
...
errors.errors = { /* error code */ }`